server.js

```diff
@@ -46,6 +45,37 @@ function toLocation(geoData) {
     };
 }
 
+app.get('/weather', (request, response) => {
+    try {
+        const weather = request.query.weather;
+        const result = getWeather(weather);
+        response.status(200).json(result);
+    }
+    catch(err) {
+        response.status(500).send('Oops, something went wrong please try again.');
+    }
+});
+
+const weather = require('./data/darksky.json');
+
+function getWeather(/*weather*/) {
+    //api call will go here
+    return toWeather(weather);
+}
+
+function toWeather(weather) {
+    const pathToData = weather.daily.data;
+    let dataArray = [];
+    for(let i = 0; i < pathToData.length; i++) {
```

Comment:
instead of doing a for loop here, you could pathToData.map(   **return object for each i **)